### PR TITLE
Enable clippy check

### LINKED ISSUE #110: Clippy should not fail in CI
Add `#[allow]` to ignore the errors in generated files.

### PR DESCRIPTION
**Pull request by chenyukang**
Fixes #110

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CLIPPY_OPTIONS := -- -D warnings -D clippy::redundant_clone -D clippy::enum_glob_use \
```
Comment (contrun) [inline Makefile]: Using a config file is more preferable https://doc.rust-lang.org/stable/clippy/configuration.html.

File: src/ckb/mod.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::all)]
 pub mod config;
```
Comment (contrun) [inline src/ckb/mod.rs]: Is this added unintentionally? This file is not automatically generated.

File: src/ckb/schema/gen.sh [hunk 3]
```diff
@@ -12,4 +12,9 @@ for file in "${files[@]}"; do
     f="$schema_dir/$file"
     output_file="$gen_dir/$(basename "${f%%.mol}.rs")"
     "$MOLC" --language rust --schema-file "$f" | rustfmt > "$output_file"
+
+    ## ignore them in clippy
+    DIRECTIVE="#![allow(clippy::all)]"
+    echo -e "$DIRECTIVE\n$(cat "$output_file")" > "$output_file"
```
Comment (contrun) [inline src/ckb/schema/gen.sh]: Reading and writing the same file in a single shell pipeline may cause problems like incomplete read. It is normally recommended to use sponge https://man.archlinux.org/man/sponge.1 . But here we can just replace the cat command with the above molc command.

File: src/ckb/channel.rs [hunk 4]
```diff
@@ -1873,6 +1867,8 @@ impl ChannelActorState {
                     "Setting local_committed_at for tlc {:?} to commitment number {:?}",
                     tlc.tlc.id, commitment_numbers
                 );
+                // FIXME: need to confirm with @contrun
+                #[allow(clippy::if_same_then_else)]
                 if is_received {
                     tlc.creation_confirmed_at = Some(commitment_numbers);
                 } else {
```
Comment (chenyukang) [inline src/ckb/channel.rs]: @contrun please confirm this, is this unintentionally?

File: src/ckb/schema/gen.sh (hunk 3 above)
Comment (chenyukang) [reply src/ckb/schema/gen.sh]: `sponge` is not a default tool on macos, fixed it with `mktemp`.

File: src/ckb/channel.rs (hunk 4 above)
Comment (contrun) [reply src/ckb/channel.rs]: The is_received check should be deleted.

**Review by contrun — APPROVED** [review]

File: src/ckb/schema/gen.sh (hunk 3 above)
Comment (quake) [reply src/ckb/schema/gen.sh]: please add `#[allow(clippy::all)]` to the gen/mod.rs thus we can keep the generated file unchanged.
```
#[allow(clippy::all)]
pub mod cfn;
```

File: src/ckb/gen/cfn.rs [hunk 5]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::all)]
```
Comment (doitian) [inline src/ckb/gen/cfn.rs]: https://github.com/nervosnetwork/cfn-node/blob/d1cdb4f2464beab34e962e6b1b8aeac85b40d74b/src/ckb/gen/mod.rs#L2 does not work?

**Review by doitian — APPROVED** [review]

File: src/ckb/gen/cfn.rs (hunk 5 above)
Comment (chenyukang) [reply src/ckb/gen/cfn.rs]: yes, it works to add another `#[allow(clippy::all)]`.

**Pull request merged**